packages/kbn-unified-data-table/src/utils/get_render_cell_value.tsx

```diff
@@ -98,27 +128,8 @@ export const getRenderCellValueFn = ({
       );
     }
 
-    /**
-     * when using the fields api this code is used to show top level objects
-     * this is used for legacy stuff like displaying products of our ecommerce dataset
-     */
-    const useTopLevelObjectColumns = Boolean(
-      useNewFieldsApi &&
-        !field &&
-        row?.raw.fields &&
-        !(row.raw.fields as Record<string, unknown[]>)[columnId]
-    );
-
     if (isDetails) {
-      return renderPopoverContent({
-        row,
-        field,
-        columnId,
-        dataView,
-        useTopLevelObjectColumns,
-        fieldFormats,
-        closePopover,
-      });
+      return popoverContent;
```

Comment:
Looking at the code this seems be changing when it's being executed. Ideally it should be only when user triggers the popover to open (`isDetails` gets true) instead of when a cell itself is rendered.

Reply:
Thanks for pointing it out. I think now that It was an unnecessary change so I have reverted it.